packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -59,7 +59,25 @@ class IOSDevices extends PollingDeviceDiscovery {
   @override
   bool get requiresExtendedWirelessDeviceDiscovery => true;
 
-  StreamSubscription<Map<XCDeviceEvent, String>>? _observedDeviceEventsSubscription;
+  StreamSubscription<XCDeviceEventNotification>? _observedDeviceEventsSubscription;
+  final Map<String, IOSDevice> _cachedPolledDevices = <String, IOSDevice>{};
+
+  /// Maps device id to a map of the device's observed connections. When the
+  /// mapped connection is `true`, that means that observed events indicated
+  /// the device is connected via that particular interface.
+  ///
+  /// The device id must be missing from the map or both interfaces must be
+  /// false for the device to be considered disconnected.
+  ///
+  /// Example:
+  /// {
+  ///   device-id: {
+  ///     usb: false,
+  ///     wifi: false,
+  ///   },
+  /// }
+  final Map<String, Map<XCDeviceEventInterface, bool>> _observedConnectionsForDevice =
```

Comment:
nit: "by" tends to indicate what the key is representing, in this case the device identifier
```suggestion
  final Map<String, Map<XCDeviceEventInterface, bool>> _observedConnectionsByDeviceId =
```